### PR TITLE
Added WebSocket arg to allow manually setting port

### PR DESCRIPTION
**Pull request by ijjk**
Saw a reply on the original pull request that the WebSocket using a random port broke their set up so I added a `--websocket` or `-w` argument similar to the `-p` argument to allow manually setting this port also.

### CODE REVIEW

**Comment by timneutkens** [comment]
Could someone that ran into this explain exactly why this wouldn't work? We're using Node.js defaults by listening on port 0. So it seems like the correct default / the user shouldn't need to worry about this 🤔

**Comment by timneutkens** [comment]
Potentially because of a docker-compose type environment I guess?

**Comment by ijjk** [comment]
Yeah, I would think using a random port would work for most setups using Next. Although being able to manually map it could be good in case anyone keeps their environment's firewall locked down on a port basis or they are using a docker-compose environment as you said.

@glenarama could you explain a little more your use case for manually mapping the WebSocket port?

**Comment by glenarama** [comment]
I'm using a Chromebook for Dev, it has a Linux container which I use for Dev. 
In order to do local in browser testing/dev I need to open up specific ports. Currently the page reloads every few seconds because the port is blocked.

**Review by timneutkens — CHANGES_REQUESTED** [review]
I think this should only be a `next.config.js` option. I'm not sure about the name though 🤔 `devWebsocketPort` 🤔

**Comment by ijjk** [comment]
The `devWebSocketPort` name seems to make sense, I just pushed up that change.

**Comment by timneutkens** [comment]
@ijjk could you add a section to the readme explaining that by default there is no need to configure this, but if you do you can do it by setting `devWebSocketPort`

**Comment by timneutkens** [comment]
Could you also add a test for this if possible. There's a test suite for config options in `test/integration/config`

**Comment by ijjk** [comment]
I added a note in the README under "Configuring the onDemandEntries" let me know if that's not the right place for it. I also added a test in the suite you mentioned. I had to dynamically write to the `next.config.js` for that test suite since there is the possibility of the port I define in the config being unavailable.

File: packages/next/server/hot-reloader.js [hunk 1]
```diff
@@ -173,8 +173,9 @@ export default class HotReloader {
     await this.clean()
 
     this.wsPort = await new Promise((resolve, reject) => {
+      const { devWebSocketPort } = this.config
       // create dynamic entries WebSocket
-      this.wss = new WebSocket.Server({ port: 0 }, function (err) {
+      this.wss = new WebSocket.Server({ port: devWebSocketPort || 0 }, function (err) {
```
Comment (timneutkens) [inline packages/next/server/hot-reloader.js]: Let's always read from `devWebsocketPort` and add `0` as a default in packages/next-server/server/config.js` Then we have a single source of truth 👍

Comment (ijjk) [reply packages/next/server/hot-reloader.js]: Good idea, I updated it

**Review by timneutkens — APPROVED** [review]

File: packages/next/README.md [hunk 2]
```diff
@@ -1292,7 +1292,9 @@ module.exports = {
     maxInactiveAge: 25 * 1000,
     // number of pages that should be kept simultaneously without being disposed
     pagesBufferLength: 2,
-  }
+  },
+  // optionally configure a port for the onDemandEntries WebSocket, not needed by default
+  devWebSocketPort: 3001,
```
Comment (timneutkens) [inline packages/next/README.md]: I just realized we have a `onDemandEntries` key, totally forgot about it 🤦‍♂️ Can you move `devWebSocketPort` under `onDemandEntries` and call it `websocketPort` 🕵️

Comment (timneutkens) [reply packages/next/README.md]: Then I'll merge 💯

**Review by timneutkens — CHANGES_REQUESTED** [review]
As per my updated comment, sorry for changing it yet again 😅

**Comment by ijjk** [comment]
It's all good, that seems like a better place for it. I also moved the `onDemandEntries` config to the `defaultConfig` to keep that as the source of truth.

**Comment by glenarama** [comment]
Thanks guys, really appreciate it!!

**Review by timneutkens — APPROVED** [review]
Awesome thanks so much 🙏 Happy new year!

**Comment by ijjk** [comment]
Happy New Year to you too!

**Pull request merged**